kubeflow/components/filter-paraphrase.yaml

```diff
@@ -0,0 +1,136 @@
+name: Filter Paraphrases
+description: |
+  Filter paraphrases using a parser
+inputs:
+  - {name: image, description: '', default: ''}
+  - {name: s3_bucket, description: '', default: ''}
+  - {name: owner, description: '', default: ''}
+  - {name: task_name, description: '', default: '' }
+  - {name: project, description: '', default: ''}
+  - {name: experiment, description: '', default: ''}
+  - {name: dataset, description: '', default: ''}
+  - {name: s3_input_datadir, description: '', default: ''}
+  - {name: filtering_model, description: '', default: ''}
+  - {name: paraphrasing_model, description: '', default: ''}
+  - {name: paraphrase_subfolder, description: '', default: ''}
+  - {name: additional_args, description: '', default: ''}
+outputs:
+  - {name: s3_output_datadir}
+implementation:
+  container:
+    image: '{{inputs.parameters.image}}'
+    command:
+    - /bin/bash
+    - -c
+    - |
+      . /opt/genie-toolkit/lib.sh
+
+      parse_args "$0" "image s3_bucket owner task_name project experiment dataset s3_input_datadir s3_output_datadir filtering_model paraphrasing_model task_name paraphrase_subfolder" "$@"
+      shift $n
+      
+      set -e
+      set -x
+      
+      /opt/genie-toolkit/sync-repos.sh 
+
+      aws s3 sync --no-progress "${s3_input_datadir}" input_dataset/
+      aws s3 sync --no-progress --exclude '*/dataset/*' --exclude '*/cache/*' --exclude 'iteration_*.pth' --exclude '*_optim.pth' "${filtering_model}" filtering_model/
+      aws s3 sync --no-progress --exclude '*checkpoint*' "${paraphrasing_model}" paraphraser/
+
+      mkdir -p output_dataset
+      cp -r input_dataset/* output_dataset
+      
+      s3_output_dir="$s3_input_datadir" # filtering does not create a new folder
```

Comment:
Filtering needs to create a new folder, otherwise caching will not work: a change in the inputs to this step (eg a different filtering model) will not cause a change in the output, so later stages will not be rerun.